Replace `addClass` with `classList.add`

p8 opened the pull request:
This pull request includes a

- [ ] Bug fix
- [ ] New feature
- [ ] Translation

The following changes were made
- This replaces jQuery's `addClass` function with the vanilla JavaScript `classList.add`.

This is related to: https://github.com/select2/select2/issues/6196

kevin-brown commented:
Can you rebase this with the latest changes on develop (I merged in your PR to fix the CI, thank you!) and back out the `dist` changes for me? The `dist` changes cause annoying merge conflicts across PRs so I prefer to recompile it separately after I merge batches of changes.

p8 committed:
Replace `addClass` with `classList.add`

This replaces jQuery's `addClass` function with the vanilla JavaScript
`classList.add`.

p8 commented:
@kevin-brown I've rebased and removed the dist changes.
